src/App.tsx

```diff
@@ -15,36 +16,111 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  Alphaberticaly = 'Sort alphabetically',
+  Length = 'Sort by length',
+  Reverse = 'Reverse',
+}
+
+const SORT_FIELD_ALPHABET: SortType = SortType.Alphaberticaly;
+const SORT_FIELD_LENGTH: SortType = SortType.Length;
+
+function getPreparedGoods(
+  goods: string[],
+  { sortField }: { sortField: SortType },
+): string[] {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    switch (sortField) {
+      case SORT_FIELD_ALPHABET:
+        preparedGoods.sort((a, b) => a.localeCompare(b));
+        break;
+      case SORT_FIELD_LENGTH:
+        preparedGoods.sort((a, b) => a.length - b.length);
+        break;
+      default:
+        break;
+    }
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<SortType | ''>('');
+
+  let visibleGoods = sortField
+    ? getPreparedGoods(goodsFromServer, { sortField })
+    : [...goodsFromServer];
+
+  const [reversed, setReversed] = useState(false);
+
+  if (reversed) {
+    visibleGoods = visibleGoods.reverse();
+  }
+
+  const buttonClassAplhabet = classNames('button', 'is-info', {
+    'is-light': sortField !== SORT_FIELD_ALPHABET,
+  });
+
+  const buttonClassLength = classNames('button', 'is-success', {
+    'is-light': sortField !== SORT_FIELD_LENGTH,
+  });
+
+  const buttonClassReverse = classNames('button', 'is-warning', {
+    'is-light': !reversed,
+  });
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={buttonClassAplhabet}
+          onClick={() => setSortField(SORT_FIELD_ALPHABET)}
+        >
           Sort alphabetically
```

Comment:
[UX] - It's good that you're changing the button class based on the active sort field, but consider adding an `aria-pressed` attribute or similar to convey the active state to assistive technologies.